From 7afdfbae7390ed5aa465df09ca469cb23b0169a1 Mon Sep 17 00:00:00 2001
From: Nathan Regner <nathanregner@gmail.com>
Date: Sat, 30 Dec 2023 15:44:33 -0700
Subject: [PATCH] Add default klipper.elf.hex target

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 10615726..c679eb99 100644
--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,7 @@ CFLAGS_klipper.elf = $(CFLAGS) -Wl,--gc-sections
 CPPFLAGS = -I$(OUT) -P -MD -MT $@
 
 # Default targets
-target-y := $(OUT)klipper.elf
+target-y := $(OUT)klipper.elf $(OUT)klipper.elf.hex
 
 all:
 
-- 
2.42.0

